applications/GeoMechanicsApplication/python_scripts/geomechanics_T_solver.py

```diff
@@ -0,0 +1,192 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.GeoMechanicsApplication as KratosGeo
+import KratosMultiphysics.StructuralMechanicsApplication as KratosStructure
+
+# Import base class file
+from KratosMultiphysics.GeoMechanicsApplication.geomechanics_U_Pw_solver import UPwSolver
+
+def CreateSolver(model, custom_settings):
+    return TSolver(model, custom_settings)
+
+class TSolver(UPwSolver):
+    '''Solver for the solution of displacement-pore pressure coupled problems.'''
+
+    def __init__(self, model, custom_settings):
+        super().__init__(model, custom_settings)
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Construction of Solver finished.")
+
+    @classmethod
+    def GetDefaultParameters(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "solver_type": "geomechanics_T_solver",
+            "model_part_name": "PorousDomain",
+            "domain_size": 2,
+            "start_time": 0.0,
+            "model_import_settings":{
+                "input_type": "mdpa",
+                "input_filename": "unknown_name"
+            },
+            "material_import_settings" :{
+                "materials_filename": ""
+            },
+            "time_stepping": {
+                "end_time" : 1.0,
+                "time_step": 0.1
+            },
+            "buffer_size": 2,
+            "echo_level": 0,
```

Comment:
Currently we have a new flag "rebuild_level" in UP solver. You can skip rebuilding stiffness matrix in each iteration. Perhaps, you can add here, if you want.

Reply:
Added     "rebuild_level" : 2
I found it does not work for 0 and 1 in my case.